react-app/src/components/property/PropertyDialog.tsx

```diff
@@ -262,44 +263,149 @@ export const PropertyAssessedValueEditDialog = (props: IPropertyAssessedValueEdi
+      .filter((evaluation) => evaluation.Value != null)
+      .map((evaluation) => ({
+        ...evaluation,
+        BuildingId: (evaluation as BuildingEvaluation).BuildingId,
+        ParcelId: (evaluation as ParcelEvaluation).ParcelId,
+        Value: parseFloat(String(evaluation.Value)),
+        EvaluationKeyId: 0,
+        Year: evaluation.Year,
+      }));
+  };
+
+  useEffect(() => {
+    if (!initialValues?.Evaluations || initialValues.Evaluations.length === 0) {
+      assessedFormMethods.reset({ Evaluations: defaultParcelValues });
+    } else {
+      assessedFormMethods.reset({
+        Evaluations: initialValues?.Evaluations?.map((evalu) => ({
+          ...evalu,
+          Value: evalu.Value.replace(/[$,]/g, ''), // TODO: Consider some more robust handling for this at the TypeORM level.
+        })),
+        RelatedBuildings: initialRelatedBuildings?.map((building) => ({
+          Id: building.Id,
+          Evaluations: building.Evaluations?.map((evalu) => ({
+            ...evalu,
+            Value: String(evalu.Value).replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
+          })).sort((a, b) => b.Year - a.Year),
+        })),
+      });
+    }
+  }, [initialValues, initialRelatedBuildings]);
+  if (!initialValues || Object.keys(initialValues).length === 0) {
+    return null; // Or any other JSX to handle the case of empty initialValues
+  }
   return (
     <ConfirmDialog
       confirmButtonProps={{ loading: submittingParcel || submittingBuilding }}
       title={'Edit assessed values'}
       open={open}
       onConfirm={async () => {
+        const isValid = await assessedFormMethods.trigger();
+        if (!isValid) {
+          return;
+        }
         const formValues = assessedFormMethods.getValues();
-        const evalus = { Id: initialValues.Id, PID: initialValues.PID, ...formValues };
+        const evalus = {
+          Id: initialValues.Id,
+          PID: initialValues.PID,
+          Evaluations: evaluationMapToRequest(formValues.Evaluations),
+        };
         if (propertyType === 'Parcel') {
-          await submitParcel(initialValues.Id, evalus);
-          for (const building of formValues.RelatedBuildings) {
-            if (building.Evaluations.length) {
-              await submitBuilding(building.Id, building);
-            }
+          const parcelUpdateResponse = await submitParcel(initialValues.Id, evalus);
+          if (formValues.RelatedBuildings) {
+            const buildingUpdatePromises = formValues.RelatedBuildings.map(async (building) => {
+              const updatedBuilding: Partial<Building> = {
+                ...building,
+                Evaluations: evaluationMapToRequest(building.Evaluations),
+              };
+              await submitBuilding(building.Id, updatedBuilding); // Update the building
+            });
+            await Promise.all(buildingUpdatePromises);
           }
+          await parcelUpdateResponse;
```

Comment:
What is this line doing? It looks like the value would have been resolved on line 336.